### PR TITLE
Adding a Skin Manager

### PR DESCRIPTION
**Pull request by tacchinotacchi**
Obviously not ready yet, PR'd for conceptual reasons.

### CODE REVIEW

File: osu.Game/osu.Game.csproj [hunk 1]
```diff
@@ -289,6 +290,10 @@
       <Project>{d9a367c9-4c1a-489f-9b05-a0cea2b53b58}</Project>
       <Name>osu.Game.Resources</Name>
     </ProjectReference>
+    <ProjectReference Include="..\osu-framework\osu.Framework.Desktop\osu.Framework.Desktop.csproj">
```
Comment (peppy) [inline osu.Game/osu.Game.csproj]: ????

File: osu.Game/Configuration/SkinManager.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+using osu.Framework.Graphics.Textures;
+using osu.Framework.IO.Stores;
+using osu.Framework.Platform;
+
+namespace osu.Game.Configuration
+{
+    public class SkinManager
+    {
+        public static Skin DEFAULT_SKIN = new Skin {
            Name = @"default",
```
Comment (peppy) [inline osu.Game/Configuration/SkinManager.cs]: may want to check your editor's line endings

**Comment by peppy** [comment]
conceptually this is the right direction, but each skin should also have resource stores. the skin"manager" would add the skin's stores to its own as required.

File: osu.Game/Configuration/SkinManager.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        private Skin selected;
+
+        public SkinManager()
+        {
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(BasicGameHost host, OsuConfigManager config)
+        {
+            selected = config.Get<Skin>(OsuConfig.Skin);
```
Comment (peppy) [inline osu.Game/Configuration/SkinManager.cs]: get a bindable here and use it to handle skin changes

**Comment by smoogipooo** [comment]
I'm going to close this for now, pending us deciding how skinning is going to work (this is a little too early at this point).

**Pull request closed without merging**